### PR TITLE
[WIP] Fix delete for VMSS flex

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Fix delete for VMSS flex
```

### CODE REVIEW

File: azure/services/scalesetvms/scalesetvms.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func (s *Service) deleteVMSSFlexVM(ctx context.Context, resourceID string) error
 	resourceGroup := parsed.ResourceGroup
 	resourceName := strings.TrimPrefix(s.Scope.ProviderID(), azure.ProviderIDPrefix)
 	resourceNameSplits := strings.Split(resourceName, "/")
-	resourceName = resourceNameSplits[len(resourceNameSplits)-3] + "_" + resourceNameSplits[len(resourceNameSplits)-1]
+	resourceName = resourceNameSplits[len(resourceNameSplits)-1]
```
Comment (CecileRobertMichon) [inline azure/services/scalesetvms/scalesetvms.go:158]: /hold

I'll follow up with unit tests, code cleanup, etc. but for now just making sure this fixes the e2e test

**Comment by mboersma** [comment]
The `-optional` test passed the Flex spec (failed on clusterclass), so this fix works. 💜

File: azure/services/scalesetvms/scalesetvms.go (hunk 1 above)
Comment (jackfrancis) [reply azure/services/scalesetvms/scalesetvms.go:158]: passed!

**Comment by CecileRobertMichon** [comment]
/close

@mboersma will open a new PR with cleaner fix and tests

**Pull request closed without merging**